### PR TITLE
nav2_smac_planner: handle corner case where start and goal are on the same cell

### PR DESCRIPTION
**Pull request by DylanDeCoeyer-Quimesis**
This case was already properly handled in the smac_planner_2d (since PR #2488), but it was still leading to an A* backtrace failure in the smac_planner_hybrid and smac_planner_lattice. Let's harmonize the handling of this case.

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #4792 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on |  |
| Does this PR contain AI generated software? | No |

---

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_smac_planner/src/smac_planner_hybrid.cpp [hunk 1]
```diff
@@ -404,6 +414,20 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
...
 
+  // Corner case of start and goal being on the same cell
+  if (std::floor(mx_start) == std::floor(mx_goal) &&
+    std::floor(my_start) == std::floor(my_goal))
+  {
+    pose.pose = start.pose;
+    // if we have a different start and goal orientation, set the unique path pose to the goal
+    // orientation
+    if (start.pose.orientation != goal.pose.orientation) {
+      pose.pose.orientation = goal.pose.orientation;
```
Comment (SteveMacenski) [inline nav2_smac_planner/src/smac_planner_hybrid.cpp:425]: Why not just always do this? I don't think we even need the if branch

File: nav2_smac_planner/src/smac_planner_hybrid.cpp [hunk 2]
```diff
@@ -404,6 +414,20 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
...
+    }
+    plan.poses.push_back(pose);
+    return plan;
```
Comment (SteveMacenski) [inline nav2_smac_planner/src/smac_planner_hybrid.cpp:428]: Please check on debug visualizations parameter and publish the plan to the topic. Some people strongly rely on that

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Otherwise, LGTM. Make sure to do these same comments in the State Lattice planner & add into the unit tests a plan where the start and goal are the same bin to exercise the change in unit testing

File: nav2_smac_planner/src/smac_planner_hybrid.cpp (hunk 1 above)
Comment (DylanDeCoeyer-Quimesis) [reply nav2_smac_planner/src/smac_planner_hybrid.cpp]: I took inspiration from the structure in smac_planner_2d, but you're right, there is no added value here.

File: nav2_smac_planner/src/smac_planner_hybrid.cpp (hunk 2 above)
Comment (DylanDeCoeyer-Quimesis) [reply nav2_smac_planner/src/smac_planner_hybrid.cpp:431]: I noticed that the plan is published independently from the value of `debug_visualizations`. So I did the same in this case.

File: nav2_smac_planner/test/test_smac_2d.cpp [hunk 3]
```diff
@@ -77,6 +77,15 @@ TEST(SmacTest, test_smac_2d) {
   } catch (...) {
   }
 
+  // corner case where the start and goal are on the same cell
+  goal.pose.position.x = 0.01;
+  goal.pose.position.y = 0.01;
+
+  try {
+    planner_2d->createPlan(start, goal, dummy_cancel_checker);
```
Comment (SteveMacenski) [inline nav2_smac_planner/test/test_smac_2d.cpp:85]: This shouldn't throw, right? We should get a single-point path that we need to check in the return for with `EXPECT_TRUE` on the `size()`

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Otherwise, LGTM. Test just needs to actually test the result of the exercised bit

**Review by SteveMacenski — APPROVED** [review]
Great work, thanks!
